compose/config/fields_schema.json

```diff
@@ -37,7 +37,15 @@
 
         "environment": {
```

Comment:
I think these constraints here are the same ones we'd want for `labels`, and most likely for `extra_hosts` as well (where are the two places that `list_or_dict` is used).

What do you think about moving this to be the new definition of `list_or_dict` ?

If we do that, I think we'd want to change the format to something like `non_bool_scalar`